pkgs/development/python-modules/tree-sitter-grammars/default.nix

```diff
@@ -0,0 +1,140 @@
+                  return PyLong_FromVoidPtr(${snakeCaseName}());
+              }
+
+              static PyMethodDef methods[] = {
+                  {"language", _binding_language, METH_NOARGS,
+                  "Get the tree-sitter language for this grammar."},
+                  {NULL, NULL, 0, NULL}
+              };
+
+              static struct PyModuleDef module = {
+                  .m_base = PyModuleDef_HEAD_INIT,
+                  .m_name = "_binding",
+                  .m_doc = NULL,
+                  .m_size = -1,
+                  .m_methods = methods
+              };
+
+              PyMODINIT_FUNC PyInit__binding(void) {
+                  return PyModule_Create(&module);
+              }
+            '';
+            destination = "/${snakeCaseName}/binding.c";
+          })
+          (pkgs.writeTextFile {
+            name = "${drvPrefix}-setup.py";
+            text = ''
+              from platform import system
+              from setuptools import Extension, setup
+
+
+              setup(
+                name="${snakeCaseName}",
+                version="${version}",
+                packages=["${snakeCaseName}"],
+                ext_package="${snakeCaseName}",
+                ext_modules=[
+                  Extension(
+                    name="_binding",
+                    sources=["${snakeCaseName}/binding.c"],
+                    extra_objects = ["${grammarDrv}/parser"],
+                    extra_compile_args=(
+                      ["-std=c11"] if system() != 'Windows' else []
+                    ),
+                    define_macros=[
+                      ("Py_LIMITED_API", "0x03080000"),
+                      ("PY_SSIZE_T_CLEAN", None)
+                    ],
+                    py_limited_api=True,
+                  )
+                ],
+              )
+            '';
+            destination = "/setup.py";
+          })
+          (pkgs.writeTextFile {
+            name = "${drvPrefix}-test";
+            text = ''
+              from ${snakeCaseName} import language
+              from tree_sitter import Language, Parser
+
+
+              def test_language():
+                lang = Language(language(), "${langIdent}")
+                assert lang is not None
+                parser = Parser()
+                parser.set_language(lang)
+                tree = parser.parse(bytes("", "utf-8"))
+                assert tree is not None
+            '';
+            destination = "/tests/test_language.py";
+          })
+        ];
+      };
+
+      preCheck = ''
+        rm -r ${snakeCaseName}
+      '';
+
+      nativeCheckInputs = [ python.pkgs.tree-sitter pytestCheckHook ];
+      pythonImportsCheck = [ snakeCaseName ];
+
+      meta = {
+        description = "Python bindings for ${langIdent} tree-sitter grammar";
+        maintainers = with lib.maintainers; [ a-jay98 adfaure mightyiam stepbrobd ];
+        license = lib.licenses.mit;
+      };
+    };
+in
+# TODO pkgset or flattened?
```

Comment:
We didn't see any pkgsets inside python3Packages so we will go ahead and flatten this unless we're told otherwise. But we didn't any flattening in top-level either so we guess this will be the first.

Reply:
We changed our minds. We saw at least two package sets nested in `python3Packages`. We saw `bootstrap` and `qt6`. And this one is literally mapped from an existing package set (`tree-sitter.builtGrammars`) so it seems appropriate.